test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
+      ex.message must include("internal server error")
+
     }
 
-    "result in a InternalServerError exception" when {
-      "when a InternalServerError http response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any()))
-          .thenReturn(Future.successful(InternalServerErrorHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[InternalServerException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"internal server error\""
-      }
+    "should return a BadRequestException when BAD_REQUEST response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(BAD_REQUEST).withBody("bad request")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[BadRequestException] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("bad request")
+
     }
 
-    "result in a Locked exception" when {
-      "when a Locked response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(LockedHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[LockedException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"locked\""
-      }
+    "should return a LockedException when LOCKED response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(LOCKED).withBody("locked")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[Upstream4xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("locked")
+
     }
 
-    "result in an HttpException" when {
-      "when a unknown error http response is received from the http get call" in {
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(UnknownErrorHttpResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[HttpException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"unknown response\""
-      }
+    "should return a UnauthorizedException when UNAUTHORIZED response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(UNAUTHORIZED).withBody("unauthorized")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[UnauthorizedException] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("unauthorized")
+
     }
 
-    "result in an UnauthorisedException" when {
-      "when an Unauthorised response is received from the http get call" in {
-        val unauthorisedResponse =
-          HttpResponse(UNAUTHORIZED, Some(JsString("unauthorised response")), Map("ETag" -> Seq("34")))
+    "should return a HttpException when unknown response" in {
```

Comment:
This is not quite true 4xx statuses will throw a Upstream4xxResponse

Reply:
Changed its name to Upstream4xxResponse